### ISSUE TITLE
Add sample docker-compose.yml

### ISSUE THREAD

**Issue by yariksheptykin** [pimcore/skeleton#12]
Hi guys!

This is a discussion bug.

What are your thoughs on adding a `docker-compose.yml` to the pimcore project skeleton?

Docker and docker-compose have meanwhile become standard tools for software developers. Currently to get this skeleton project up and running one needs to: a) issue `composer create-project` to download this project template, and b) set up supporting services (redis, mysql, elasticsearch, etc) to be able to install and run pimcore application. To deal with the latter a casual developer, such as myself, would use `docker` and `docker-compose` to provision the execution enviroment for local development. Often projects such as https://github.com/dpfaffenbauer/pimcore-docker-compose would be used as a template to bootstrap this process.

My proposal ist to add a sample docker-compose.yaml to this skeleton as a reference for setting up a vanilla developer environment. We can all benefit from this it two ways:
 1. saving us from doing a necessary step b), and
 2. building up a shared knowledge of system setup encoded in a docker-compose.yml and Dockerfiles.

docker-compose.yml from (https://github.com/dpfaffenbauer/pimcore-docker-compose) seems to me a reasonalble starting point for such a shared docker-compose.yml.
What do you think, @dpfaffenbauer ?

**Comment by dpfaffenbauer** [comment]
I am down :)

**Comment by yariksheptykin** [comment]
> I am down :)

Thanks for the quick reply @dpfaffenbauer . What do you mean by "being down"? Do you mean you dont like the idea (like "Thumbs Down")? Or not interested in the discussion?

**Comment by dpfaffenbauer** [comment]
That means I am in. Thumbs up basically

**Comment by yariksheptykin** [comment]
How about you @brusch , @fashxp ? What are your thoughts on that?

**Comment by brusch** [comment]
@yariksheptykin would be fine for me, PRs are welcome :)

**Comment by yariksheptykin** [comment]
@brusch , @dpfaffenbauer , could you take a look on Luka's MR? Thats basically an identical copy of @dpfaffenbauer 's composer file including all the git history. What do you think about it?

**Comment by yariksheptykin** [comment]
@brusch could you take a look at Luka's MR? Is there something blocking it?

**Comment by brusch** [comment]
@yariksheptykin yep, lack of time 😉I'll try to do as soon as possible.

**Comment by yariksheptykin** [comment]
NOTE: Here is another example of skeleton shipped with `docker-compose.yml` from @dpfaffenbauer https://github.com/w-vision/pimcore-skeleton

**Comment by dpfaffenbauer** [comment]
Here another one ;)
https://github.com/dpfaffenbauer/pimcore-skeleton

**Comment by yariksheptykin** [comment]
@brusch is there anything I can do to help you with this MR? I totally understand that you are under a huge time pressure due to maintaining a project of the scope of pimcore. It is very time demanding but you guys do a great job on addressing issues and MRs quickly. Maybe we could do something to take some load from you to get the skeleton going?
Here is one idea. You could grant me permissions to step in and supervise parts of the skeleton regarding docker. We could discuss this in a short call if you think this it is a good idea. To my experience such distribution of responsibilities is a great means to reduce time pressure on individuals and increase overall engagement.

**Comment by dpfaffenbauer** [comment]
@yariksheptykin be patient. there is no need to hurry with this. I am still not sure if this is a way to go for pimcore, and I maintain the docker images..... just give it time.

**Comment by yariksheptykin** [comment]
@dpfaffenbauer yeah you right, I might be pushing too much. @brusch, sorry about that.

@dpfaffenbauer , I hope I see your point on docker images right. Distributing a compose file that relies on non-official images is risky. The best way would be to distribute own "official" pimcore images. This however would require a somewhat bigger time investment upfront. It seems to me that the core developers are short on time to address this option.
I am ready to contribute but if the integration process is slow, there is not much gain for me as our organization needs the skeleton right now for bootstrapping new pimcore projects. As I wrote in the beginning each project gets then immediately equipped with some kind of docker configuration. I started this discussion in hope to remedy this situation.

But we can go slow on this one for now, thats ok with me.

**Comment by dpfaffenbauer** [comment]
If you really need a skeleton with docker, use mine :P

I think that Pimcore should somewhat create there own images where Pimcore will be installed on startup automatically, to make bootstrapping pimcore even easier. They can rely on my images if they want to, but don't have to. 

But, thats a decision the Pimcore Team has to make.

**Comment by yariksheptykin** [comment]
Does not seem like there is any interest in this topic, so closing the issue. Thanks everyone for participation and feedback.

**Comment by dpfaffenbauer** [comment]
sorry, but that is not how OSS works. You need to give things time, they need discussion and patience to find the right solution. Just closing things and pretending there is no interest will not make things better. You never really gave Pimcore a chance to even tell us their thoughts about it....

**Comment by yariksheptykin** [comment]
Hi @dpfaffenbauer, thanks for your fair comment. To be clear: I am not pretending to have no interest, I DO have interest in the topic and ready to contribute or provide solutions. Unfortunately the speed at wthich this conversation is going is not briging me any further towards a solution, so I need to look for an alternative approach. @brusch If the topic has any relevance, feel free to re-open or start a new thread with an alternative proposal. I will be happy to contribute.

**Comment by dpfaffenbauer** [comment]
So look for an alternative, but does that stop creating a Pimcore Docker Environment directly from Pimcore? I don't think so....

**Comment by dpfaffenbauer** [comment]
https://github.com/pimcore/skeleton/pull/16

**Comment by dpfaffenbauer** [comment]
also for the demo: https://github.com/pimcore/demo/pull/161

**Comment by dpfaffenbauer** [comment]
We went a completely different approach. I talked to Pimcore and I offered them to give them my images and also maintain them within the Pimcore namespace. So the new images are basically the same as mine:

`dpfaffenbauer/pimcore:PHP7.3-apache` now is `pimcore/pimcore:PHP.3-apache`

and so on for all other tags. `pimcore/pimcore` now also includes PHP7.4 images: `pimcore/pimcore:PHP7.4-apache`, `pimcore/pimcore:PHP7.4-fpm`, `pimcore/pimcore:PHP7.4-cli`.
